File: lc383.cpp
#include <iostream>
#include <string>

using namespace std;

bool canConstruct(string ransomNote, string magazine);

int main(int argc, char** argv) {
  string ransomNote, magazine;

  // Get user input for ransom note
  cout << "Enter the ransom note: ";
  getline(cin, ransomNote);

  // Get user input for magazine
  cout << "Enter the magazine: ";
  getline(cin, magazine);

  // Call the canConstruct function and print the result
  if (canConstruct(ransomNote, magazine)) {
    cout << "Ransom note can be constructed from the magazine." << endl;
  } else {
    cout << "Ransom note cannot be constructed from the magazine." << endl;
  }

  return 0;
}

bool canConstruct(string ransomNote, string magazine) {
  int letterCount[26] = {0};
  for (char i : magazine) {
    ++letterCount[i - 'a'];
  }

  for (char i : ransomNote) {
    if (letterCount[i - 'a']--) {
      continue;
    } else {
      return false;
    }
  }

  return true;
}

File: lc383.rs
pub fn can_construct(ransom_note: String, magazine: String) -> bool {
    let mut count = vec![0; 26];
    for i in magazine.chars() {
        count[i as usize - 'a' as usize] += 1;
    }
    
    for i in ransom_note.chars() {
        count[i as usize - 'a' as usize] -= 1;
        if count[i as usize - 'a' as usize] < 0 {
            return false;
        }
    }

    true
}
